### PR TITLE
Miscellaneous changes to keep up to date with the cookiecutter

### PR DESCRIPTION
**Pull request by marcospri**
- Missing requirements
- Environment variables in tox.ini
- New programs in supervisor

Not included:

- (mostly) linter changes (migration from pylint to ruff)

### CODE REVIEW

File: .cookiecutter/includes/requirements/dev.in [hunk 1]
```diff
@@ -0,0 +1 @@
+h_testkit
```
Comment (seanh) [inline .cookiecutter/includes/requirements/dev.in:1]:
```suggestion
h-testkit
```

Should be `h-testkit` I think, but why is this needed in `dev.in`? The cookiecutter will automatically add this to `tests.in` and `functests.in`

**Review by seanh — APPROVED** [review]

File: .cookiecutter/includes/requirements/dev.in (hunk 1 above)
Comment (marcospri) [reply .cookiecutter/includes/requirements/dev.in:1]: Ah, it looks like they are both equivalent:

https://packaging.python.org/en/latest/specifications/name-normalization/#name-normalization

Change it over the `-` version for consistency :+1: 

We added this to dev because `make shell` depends on it
